_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,234 @@ also change the name of the columns in the DataFrame like this:
-value in the list is plotted at `x=0`, the second at `x=1`, etc). If the
-function `plot()` receives two lists, it assumes the first one is the x-values
-and the second the y-values. The line connecting the points will follow the list
-in order:
+### `plt` pyplot versus object based matplotlib
 
-~~~
-    plt.plot([6.8, 4.3, 3.2, 8.1], list_numbers)
-    plt.show()
-~~~
-{: .language-python}
-
-A third, optional argument in `plot()` is a string of characters that indicates
-the line type and color for the plot. The default value is a continuous blue
-line. For example, we can make the line red (`'r'`), with circles at every data
-point (`'o'`), and a dot-dash pattern (`'-.'`). Look through the matplotlib
-gallery for more examples.
+Matplotlib integrates nicely with the numpy package and can use numpy arrays as input of the available plot functions. Consider the following example data, created with numpy:
 
 ~~~
-    plt.plot([6.8, 4.3, 3.2, 8.1], list_numbers, 'ro-.')
-    plt.axis([0,10,0,6])
-    plt.show()
+import numpy
+x = numpy.linspace(0, 5, 10)
+y = x ** 2
 ~~~
 {: .language-python}
 
-The command `plt.axis()` sets the limits of the axes from a list of `[xmin,
-xmax, ymin, ymax]` values (the square brackets are needed because the argument
-for the function `axis()` is one list of values, not four separate numbers!).
-The functions `xlabel()` and `ylabel()` will label the axes, and `title()` will
-write a title above the figure.
 
-A single figure can include multiple lines, and they can be plotted using the
-same `plt.plot()` command by adding more pairs of x values and y values (and
-optionally line styles):
+To make a scatter plot of `x` and `y`, we can directly use the `plot` command:
 
 ~~~
-    import numpy as np
+plt.plot(x, y, '-')
+~~~
+{: .language-python}
 
-    # Create a numpy array between 0 and 10, with values evenly spaced every 0.5
-    t = np.arange(0., 10., 0.5)
+![Line plot of x versus y](../fig/08_line_plot.png)
 
-    # Red dashes with no symbols, blue squares with a solid line, and green triangles with a dotted line
-    plt.plot(t, t, 'r--', t, t**2, 'bs-', t, t**3, 'g^:')
+or create a matplotlib `figure` and `axis` object first and add the plot later on:
 
-    plt.xlabel('This is the x axis')
-    plt.ylabel('This is the y axis')
-    plt.title('This is the figure title')
+> ## Tip: Cross-Platform Visualization of Figures
+> Jupyter Notebooks make many aspects of data analysis and visualization much simpler. This includes
+> doing some of the labor of visualizing plots for you. But, not every one of your collaborators
+> will be using a Jupyter Notebook. The .show() command allows you to visualize plots
+> when working at the command line, with a script, or at the iPython interpreter. In the 
+> previous example, adding  `plt.show()` after the creation of the plot will enable your 
+> colleagues who aren't using a Jupyter notebook to reproduce your work on their platform.
+{: .callout}
```

Comment:
How about moving this callout box to right after when `plt.plot` command is used and before the `Line plot of x vs y`

Reply:
I tried that and it looks a bit busy. How about the way I have it now?

Reply:
looks good!